source/common/ssl/context_impl.cc

```diff
@@ -263,10 +275,14 @@ int ContextImpl::verifyCertificate(X509* cert) {
     return 0;
   }
 
-  if (!verify_certificate_hash_list_.empty() &&
-      !verifyCertificateHashList(cert, verify_certificate_hash_list_)) {
-    stats_.fail_verify_cert_hash_.inc();
-    return 0;
+  if (!verify_certificate_hash_list_.empty() || !verify_certificate_spki_list_.empty()) {
```

Comment:
nit: the boolean logic here is an eyeful, and given how important it is, is there a longer form that would be a bit more readable?

Reply:
Done.